src/neptune/metadata_containers/metadata_container.py

```diff
@@ -665,6 +667,8 @@ def _fetch_entries(self, child_type: ContainerType, query: NQLQuery, columns: Op
             columns=columns,
         )
 
+        leaderboard_entries = itertools.islice(leaderboard_entries, limit) if limit else leaderboard_entries
```

Comment:
I think that it's beneficial to still pass it to the pagination function as the limit may be much smaller than `step_size`. With the current approach, we're going to fetch for instance `100` entries even though the `limit` was set to `1`.

Reply:
You can probably just set `step_size` to a minimum of the `limit` and the old step_size. It should work as well.